### PR TITLE
Key Vault additional property and update

### PR DESCRIPTION
**Pull request by pomortaz**
1. Changing the IssuerReference to IssuerParameters
2. Adding certificateType 
3. Constraints on the maxresult for paging.

### CODE REVIEW

**Comment by pomortaz** [comment]
@rohmano please review.

**Comment by rohmano** [comment]
Looks good, signed off.

**Comment by pomortaz** [comment]
@markcowl and @hovsepm can you please merge the PR?

**Comment by pomortaz** [comment]
A test, irrelevant to my change is failing. Can you please take a look?

File: src/KeyVault/Microsoft.Azure.KeyVault/project.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "2.0.1-preview",
+  "version": "2.0.2-preview",
```
Comment (hovsepm) [inline src/KeyVault/Microsoft.Azure.KeyVault/project.json]: Please change the version in AssemblyInfo.cs file as well

**Comment by hovsepm** [comment]
LGTM :shipit:

**Pull request merged**